examples/arkode/CXX_serial/ark_pendulum.cpp

```diff
@@ -0,0 +1,421 @@
+      ARKodeButcherTable B = ARKodeButcherTable_Alloc(2, SUNFALSE);
+
+      const sunrealtype gamma = ((SUN_RCONST(3.0) + std::sqrt(SUN_RCONST(3.0)))
+                                 / SUN_RCONST(6.0));
+
+      B->A[0][0] = gamma;
+      B->A[1][0] = SUN_RCONST(1.0) - SUN_RCONST(2.0) * gamma;
+      B->A[1][1] = gamma;
+
+      B->c[0] = gamma;
+      B->c[1] = SUN_RCONST(1.0) - gamma;
+
+      B->b[0] = SUN_RCONST(0.5);
+      B->b[1] = SUN_RCONST(0.5);
+
+      B->q = 3;
+      B->p = 0;
+
+      flag = ARKStepSetTables(arkode_mem, 3, 0, B, nullptr);
+      if (check_flag(flag, "ARKStepSetTables")) return 1;
+
+      ARKodeButcherTable_Free(B);
+    }
+    else
+    {
+      // Select a Butcher table with non-negative b values
+      flag = ARKStepSetTableName(arkode_mem, "ARKODE_SDIRK_2_1_2",
+                                 "ARKODE_ERK_NONE");
+      if (check_flag(flag, "ARKStepSetTableName")) return 1;
+    }
+  }
+
+  if (fixed_h > SUN_RCONST(0.0))
+  {
+    flag = ARKStepSetFixedStep(arkode_mem, fixed_h);
+    if (check_flag(flag, "ARKStepSetFixedStep")) return 1;
+  }
+
+  /* --------------- *
+   * Advance in Time *
+   * --------------- */
+
+  // Initial time
+  sunrealtype t = t0;
+
+  // Output the initial condition and energy
+  std::ofstream outfile("ark_pendulum.txt");
+  outfile << "# vars: t u v energy energy_err\n";
+  outfile << std::scientific;
+  outfile << std::setprecision(std::numeric_limits<sunrealtype>::digits10);
+  outfile << t << " " << ydata[0] << " " << ydata[1] << " " << eng0 << " "
+          << SUN_RCONST(0.0) << std::endl;
+
+  std::cout << std::setw(5) << "step" << std::setw(24) << "t" << std::setw(24)
+            << "u" << std::setw(24) << "v" << std::setw(24) << "e"
+            << std::setw(24) << "e err" << std::endl;
+  for (int i = 0; i < 9; i++) std::cout << "--------------";
+
+  std::cout << std::endl;
+  std::cout << std::scientific;
+  std::cout << std::setprecision(std::numeric_limits<realtype>::digits10);
+  std::cout << std::setw(5) << 0 << std::setw(24) << t << std::setw(24)
+            << ydata[0] << std::setw(24) << ydata[1] << std::setw(24) << eng0
+            << std::setw(24) << SUN_RCONST(0.0);
+  std::cout << std::endl;
+
+  while (t < tf)
+  {
+    // Evolve in time
+    flag = ARKStepEvolve(arkode_mem, tf, y, &t, ARK_ONE_STEP);
+    if (check_flag(flag, "ARKStepEvolve")) break;
+
+    // Output solution and errors
+    sunrealtype eng;
+    flag = Eng(y, &eng, nullptr);
+    if (check_flag(flag, "Eng")) return 1;
+
+    sunrealtype eng_chng = eng - eng0;
+
+    /* Output to the screen periodically */
+    long int nst;
+    flag = ARKStepGetNumSteps(arkode_mem, &nst);
+    check_flag(flag, "ARKStepGetNumSteps");
+
+    if (nst % 1000 == 0)
+    {
+      std::cout << std::setw(5) << nst << std::setw(24) << t << std::setw(24)
+                << ydata[0] << std::setw(24) << ydata[1] << std::setw(24) << eng
+                << std::setw(24) << eng_chng << std::endl;
```

Comment:
As I commented in the "answers" PR, this line seems to have difficulty when running in extended precision, in that there is no spacing between the outputs.  I imagine that you could just manually add some in:
```suggestion
      std::cout << std::setw(5) << nst << "  " << std::setw(24) << t << "  " << std::setw(24)
                << ydata[0] << "  " << std::setw(24) << ydata[1] << "  " << std::setw(24) << eng
                << "  " << std::setw(24) << eng_chng << std::endl;
```